### PR TITLE
feat: makes altText changable for images in FileItem

### LINKED ISSUE #355: Make alt-text changeable in FileItem
related to: https://github.com/graasp/graasp-builder/issues/624

### PR DESCRIPTION
**Pull request by MalinSvenberg**
closes #355

### CODE REVIEW

File: src/items/FileItem.tsx [hunk 1]
```diff
@@ -120,9 +109,21 @@ const FileItem: FC<FileItemProps> = ({
...
+      item.type === ItemType.LOCAL_FILE ? getFileExtra(item.extra) : undefined;
+    const s3FileExtra =
+      item.type === ItemType.S3_FILE ? getS3FileExtra(item.extra) : undefined;
+    const { mimetype, name: originalFileName } = {
+      ...fileExtra,
+      ...s3FileExtra,
+    };
+    const altText =
+      item.type === ItemType.S3_FILE || item.type === ItemType.LOCAL_FILE
+        ? item.settings.altText || item.name
```
Comment (spaenleh) [inline src/items/FileItem.tsx:122]: To me it feels strange to have to pull that property out of the settings, when we already pull the mimetype and originalFileName from the extras ... @MalinSvenberg

**Review by spaenleh — COMMENTED** [review]
🍵 Looks good. I rebased and localized some of the variables closer to where they are used.

What do you think of my comment ?

**Review by spaenleh — APPROVED** [review]

File: src/items/FileItem.tsx [hunk 2]
```diff
@@ -120,9 +109,22 @@ const FileItem: FC<FileItemProps> = ({
...
+      item.type === ItemType.LOCAL_FILE ? getFileExtra(item.extra) : undefined;
+    const s3FileExtra =
+      item.type === ItemType.S3_FILE ? getS3FileExtra(item.extra) : undefined;
+    const {
+      mimetype,
+      name: originalFileName,
+      altText = item.name,
+    } = {
+      ...fileExtra,
+      ...s3FileExtra,
```
Comment (pyphilia) [inline src/items/FileItem.tsx:122]: Won't this fail if `fileExtra = undefined`? Same for `s3FileExtra`

File: src/items/FileItem.stories.tsx [hunk 3]
```diff
@@ -81,7 +81,7 @@ export const ImageSVG: Story = {
       type: 'file',
       description: 'my svg description',
       path: 'item-path',
-      settings: {},
+      settings: { altText: 'my svg alt text' },
```
Comment (pyphilia) [inline src/items/FileItem.stories.tsx:84]: I guess this will change if you move to extra? @spaenleh

**Review by pyphilia — CHANGES_REQUESTED** [review]
Looks okay to me, but I have one concerns for a destructuration.

File: src/items/FileItem.stories.tsx (hunk 3 above)
Comment (spaenleh) [reply src/items/FileItem.stories.tsx:84]: Yes thank you need to update that.

File: package.json [hunk 4]
```diff
@@ -50,7 +50,7 @@
     "storybook:build": "storybook build"
   },
   "dependencies": {
-    "@graasp/sdk": "1.0.0-rc1",
+    "@graasp/sdk": "github:graasp/graasp-sdk#162-add-alttext-to-settings-fileitem",
```
Comment (pyphilia) [inline package.json:53]: Update.

**Review by pyphilia — APPROVED** [review]

**Pull request merged**